lib/documentLoader.js

```diff
@@ -2,14 +2,38 @@
  * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
  */
 'use strict';
+/*eslint max-len: ["error", { "ignoreComments": true }]*/
 
 // load locally embedded contexts
 const contexts = require('./contexts');
 
+/**
+ * documentLoader is a set of methods used to dereference jsonld documents.
+ * Documents stored at urls are usually dereferenced via https requests
+ * while DIDs can be dereferenced by a variety of means.
+ * Most of these methods are there to ensure jsonld contexts are loaded
+ * from an in-memory, immutable copy of the context
+ * document to ensure content integrity.
+ * @module documentLoader
+ */
 const api = {};
 module.exports = api;
 
 api.extendContextLoader = documentLoader => {
+  /**
+   * extendContextLoader takes in a documentLoader. It then
+   * returns a function that accepts a url. If the url is in the context
```

Comment:
I think this still needs a little bikeshedding, perhaps:

"extendContextLoader extends another JSON-LD document loader. Given a document loader to extend, this method will return a new document loader that will first check for a URL in jsonld-signature's built-in context map and, if not found, it will fall back to using the passed document loader. This utility method can be used to ensure that any local, in-memory, immutable context documents provided by jsonld-signatures will be used prior to using another document loader to load other documents."

Note that there's no guarantee that any "remote source" will ever be used here, so we should strike that.